### PR TITLE
Prevent DOM onclick from firing if button is disabled

### PR DESCRIPTION
**Pull request by macandcheese**
Idea for addressing https://github.com/Esri/calcite-components/issues/319

Thoughts @AdelheidF @driskull ?

### CODE REVIEW

File: src/components/calcite-button/calcite-button.tsx [hunk 1]
```diff
@@ -123,6 +124,12 @@ export class CalciteButton {
       const elType = this.el.getAttribute("type");
       this.type = this.childElType === "button" && elType ? elType : "submit";
     }
+    this.initialOnClick = this.el.getAttribute("onclick");
```
Comment (driskull) [inline src/components/calcite-button/calcite-button.tsx]: The only issue I see is if the `onClick` is added after initialization. 

Would this approach solve that issue? https://github.com/Esri/calcite-components/pull/317/files#diff-20c32e6a69324944338fbc027b1730d2R108

**Comment by AdelheidF** [comment]
I'm OK with not changing it. I can work around it. You guys decide...

**Comment by macandcheese** [comment]
Yeah, I could go either way. It's a nice "maybe expected" thing we could add - I can't really think of a time when a disabled button SHOULD perform an onclick...  (some analytic tracking for seeing how often a user clicked on a disabled button?) so I can see how this would be a nice thing for the component to just take care of on its own @AdelheidF .

@driskull re: onclick added after initialization - yes we could... check for updated value in a different lifecycle method,  or make a proxy prop kind of thing for onclick attribute value and Watch for changes there?

**Comment by macandcheese** [comment]
@driskull @Esri/calcite-components go or no go on this?

If go, I'll add a check for onclick added after initialization and some tests around it.

**Comment by driskull** [comment]
Im leaning towards no go 👎

**Comment by macandcheese** [comment]
Fair enough - easy enough for folks to check for disabled if they need to. Closing.

**Pull request closed without merging**